### ISSUE TITLE
Ice gathering timeout, 10 seconds delay

### ISSUE THREAD

**Issue by roginvs** [DoubangoTelecom/sipml5#237]
We have a problem: if a user has several network interfaces or if user's NAT is very strange: in such cases the ICE Gathering can take 10 seconds on outgoing calls. This is not good. I added a very simple hack: to stop gathering after 2 seconds after first candidate appeared.
We can have a look at this hack here a1fae10b5b52460f4cb5fd14315bfe24cd22cd57
I am not good in sipml5 architecture, but I think that such timeout should be tunable via sipStack creation (like in sipjs).

**Comment by vfontes** [comment]
I agree, it should be tunable.

**Comment by garronej** [comment]
Good point thank you

**Comment by vferko** [comment]
Does this change not break anything under any circumstances?

**Comment by roginvs** [comment]
@vferko , I am not sure. But we are using this hack for almost two month, and everything is still fine.

**Comment by vferko** [comment]
@roginvs, thank you, this is enough for me. I assume this file ([roginvs/sipml5/release/SIPml-api.js](https://github.com/roginvs/sipml5/blob/master/release/SIPml-api.js)) implements the proposed changes..

**Comment by roginvs** [comment]
@vferko, yes, everything is right, this link holds sipml5 with this change + patch for Chrome workaround.

**Comment by viniciusstroher** [comment]
when i use that solution in my app, some times attended transfer getting one way audio.

**Comment by SeiKiMo** [comment]
very thank you
